packages/atlas/src/views/studio/VideoWorkspace/VideoWorkspace.tsx

```diff
@@ -34,17 +40,42 @@ export const VideoWorkspace: React.FC = React.memo(() => {
   const { openWarningDialog } = useDisplayDataLostWarning()
 
   const handleVideoWorkspaceSubmit = useHandleVideoWorkspaceSubmit()
+  const handleNFTWorkspaceSubmit = () => null
+
+  const onSetFormStatus = useCallback((data: FormStatus) => {
+    if (!data) {
+      return
+    }
+    setFormStatus((prevState) => ({ ...prevState, ...data }))
+  }, [])
 
   const handleSubmit = useCallback(
     (data: VideoFormData) => {
       if (isIssuedAsNFT) {
-        setNFTCurrentStepIdx((step) => (step === null ? 0 : step + 1))
-        setVideoFormDataForNFT(data)
+        if (nftCurrentStepIdx === -1) {
+          setVideoFormDataForNFT(data)
+        }
+        if (nftCurrentStepIdx === 2) {
+          handleVideoWorkspaceSubmit(data)
+          return
+        }
+        setNftCurrentStepIdx((step) => (step === null ? 0 : step + 1))
       } else {
         handleVideoWorkspaceSubmit(data)
       }
     },
-    [handleVideoWorkspaceSubmit, isIssuedAsNFT]
+    [nftCurrentStepIdx, handleVideoWorkspaceSubmit, isIssuedAsNFT]
+  )
+
+  const handleNFTSubmit = useCallback(
```

Comment:
There is a lot of variables still using the old casing. Could you run a global replace of `NFT` -> `Nft`? (There are still some variables starting with `NFT` so they will require manual update)